src/index.ts

```diff
@@ -34,25 +34,24 @@ export async function resolve(
   context: ResolveContext,
   defaultResolve: Function,
 ): Promise<ResolveResponse> {
-  let { parentURL = baseURL } = context;
+  const { parentURL = baseURL } = context;
 
-  let forceRelative = false;
+  // let forceRelative = false;
   if (TSConfig?.paths) {
     for (const tsPath of Object.keys(TSConfig.paths)) {
-      const tsPathKey = tsPath.replace('/*', '');
+      const tsPathKey = tsPath.replace('*', '');
       if (specifier.startsWith(tsPathKey)) {
-        const pathSpecifier = TSConfig.paths[tsPath][0].replace(
-          '/*',
-          specifier.split(tsPathKey)[1],
+        const tsResolvedPath = resolvePath(
+          resolvePath(baseURL, TSConfig.baseUrl!),
+          TSConfig.paths[tsPath][0].replace('*', specifier.split(tsPathKey)[1]),
         );
 
-        forceRelative = true;
+        specifier = `./${relative(
+          dirname(fileURLToPath(parentURL)),
+          tsResolvedPath,
+        )}`;
 
-        parentURL = `${
-          pathToFileURL(resolvePath(baseURL, TSConfig.baseUrl!)).href
-        }/`;
-
-        specifier = pathSpecifier;
+        break;
```

Comment:
Can't believe I missed this before, I'll be adding in proper caching soon for the resolve hook.